### PR TITLE
Specification for pull requests

### PR DESCRIPTION
**Pull request by kevgo**
@charlierudolph @allewun

### CODE REVIEW

**Comment by kevgo** [comment]
blocked by #397

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -53,6 +53,37 @@ bin/cuke [<folder>...]
 ```
 
 
+## Pull Requests
+
+Each pull request (PR) should mention the [issue](https://github.com/Originate/git-town/issues)
+it implements in the PR description:
```
Comment (allewun) [inline CONTRIBUTING.md]: 1. "resolves" sounds better than "implements" here
2. ... in its description (less wordy)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -53,6 +53,37 @@ bin/cuke [<folder>...]
...
+
+```
+implements #123
+```
+
+When merging a PR, the message for the squash commit should follow the
```
Comment (allewun) [inline CONTRIBUTING.md]: squash[ed]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -53,6 +53,37 @@ bin/cuke [<folder>...]
...
+Add a foo feature to "git bar"
+
+This allows developers to optionally foo their code before
+they finally bar it into its own feature branch.
+
+Implements #123
+```
+
+Pull requests from core developers with write access should be submitted using
+`git ship`, i.e. squash-merged.
```
Comment (allewun) [inline CONTRIBUTING.md]: not sure the " i.e. squash-merged" is necessary

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -53,6 +53,37 @@ bin/cuke [<folder>...]
...
+Pull requests from external developers should be squashed into a single commit
```
Comment (allewun) [inline CONTRIBUTING.md]: Maybe outline how they can achieve this?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -53,6 +53,37 @@ bin/cuke [<folder>...]
...
+in their feature branch, and then accepted by a core developer
```
Comment (allewun) [inline CONTRIBUTING.md]: Is the part after the comma necessary? It seems to be explaining a very basic thing, and if someone is contributing to Git Town, they should already know how that works?

File: CONTRIBUTING.md (hunk 1 above)
Comment (kevgo) [inline CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 2 above)
Comment (kevgo) [inline CONTRIBUTING.md]: changed to "squashed"

File: CONTRIBUTING.md (hunk 3 above)
Comment (kevgo) [inline CONTRIBUTING.md]: removed

File: CONTRIBUTING.md (hunk 4 above)
Comment (kevgo) [inline CONTRIBUTING.md]: Good idea. Added `git rebase -i master` example.

File: CONTRIBUTING.md (hunk 5 above)
Comment (kevgo) [inline CONTRIBUTING.md]: Okay, taken it out.

**Comment by charlierudolph** [comment]
Once we have #336, are you okay with just using it instead of having merge commits? We already have the author retained

**Comment by kevgo** [comment]
Hmm, I thought keeping a link to the original forked repo might be nice, and make it easier for the maintainer of the fork to manage their repo. But I can also see the squash-merge version work, and that one would allow us to merge PRs with multiple commits, and have a clean history. Let's give it a shot when we have that feature in!

**Comment by kevgo** [comment]
@charlierudolph @allewun This is now unblocked, and I have cleaned it up. Please take another look!

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -64,13 +64,29 @@ and notifies you via email and through status badges in pull requests
...
+This includes the [issue](https://github.com/Originate/git-town/issues) it resolves.
+
+When merging approved PRs:
+* use `git ship`
+* the message for the squashed commit should follow the
+  [formatting guidelines for commit messages](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+* [mention the closed issue](https://help.github.com/articles/closing-issues-via-commit-messages)
+in the commit body, so that the respective issue is automatically closed.
+
+Example for a commit message:
```
Comment (allewun) [inline CONTRIBUTING.md]: for -> of

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -64,13 +64,29 @@ and notifies you via email and through status badges in pull requests
...
+## Pull Requests
 
-* if possible, use Git Town to ship your pull request
-* format your commit message according to the
-  [recommended guidelines](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
-* [mention the associated issue](https://help.github.com/articles/closing-issues-via-commit-messages)
-  in the body of the commit message, so that it is automatically closed
+Each pull request (PR) should have the same (optional) description that it will have
+when committed later.
+This includes the [issue](https://github.com/Originate/git-town/issues) it resolves.
```
Comment (allewun) [inline CONTRIBUTING.md]: > ...committed later and should reference the issue it resolves.

**Comment by allewun** [comment]
:ship: after last few comments are addressed. Sorry for the late review

File: CONTRIBUTING.md (hunk 7 above)
Comment (kevgo) [inline CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 6 above)
Comment (kevgo) [inline CONTRIBUTING.md]: done

**Pull request closed without merging**